packages/app-layout/AppLayout.tsx

```diff
@@ -0,0 +1,161 @@
+import { h, Component, prop } from 'skatejs';
+
+import {
+  GenericTypes,
+  matchMedia
+} from '../_helpers';
+
+import { HamburgerButton } from './components/HamburgerButton';
+import { Drawer } from './components/Drawer';
+import { Overlay } from './components/Overlay';
+import { Nav } from './components/Nav';
+import { NavContent } from './components/NavContent';
+
+import styles from './AppLayout.scss';
+
+const DEFAULT_RESPONSIBLE_WIDTH = '640px';
+
+type AppLayoutProps = Props & EventProps;
+
+type Attrs = {
+  'drawer-visible'?: boolean,
+  'responsive-width'?: string
+};
+
+type Props = {
+  drawerVisible?: boolean,
+  responsiveWidth?: string
+};
+
+// TODO: add onDrawerChange / onNarrowChange
+type EventProps = {};
+
+type Events = {};
+
+declare global {
+  namespace JSX {
+    interface IntrinsicElements {
+      'bl-app-layout': GenericTypes.IntrinsicCustomElement<AppLayoutProps>
+      & GenericTypes.IntrinsicBoreElement<Attrs, Events>
+    }
+  }
+}
+
+export class AppLayout extends Component<AppLayoutProps> {
+
+  static get is() { return 'bl-app-layout'; }
+
+  static get props() {
+    return {
+      drawerVisible: prop.boolean( {
+        attribute: {
+          source: true
+        }
+      } ),
+      responsiveWidth: prop.string( {
+        initial: DEFAULT_RESPONSIBLE_WIDTH,
+        attribute: {
+          source: true
+        }
+      } ),
+      narrow: prop.boolean( {
+        attribute: {
+          source: true,
+        }
+      } )
+    };
+  }
+
+  drawerVisible: boolean;
+
+  narrow: boolean;
+
+  responsiveWidth: string;
+
+  connectedCallback() {
+
+    super.connectedCallback();
+
+    this.setupQueryMatchListener();
+
+  }
+
+  disconnectedCallback() {
+
+    super.disconnectedCallback();
+
+    this.matchMediaUnsubscribe();
```

Comment:
I'm following [documentation](https://github.com/skatejs/skatejs/blob/e2eaa330a34ce790bd515fa6c0d02bd7c9e23e62/docs/examples/README.md#L31) and according to [code](https://github.com/skatejs/skatejs/blob/f5ab7073f91e89003b46303e50e2f9ce3c3b8163/src/api/component.js#L221) I don't see any harm in that. Please explain and let's submit PR to skate doc.